src/Entity/Board.php

```diff
@@ -188,6 +191,21 @@ private function removePuck($puck)
         return $value;
     }
 
+    private function getFreeCells()
+    {
+        return array_values(array_filter($this->graph->vertices, function ($vertex) {
+            return empty($vertex->getValue());
+        }));
+    }
+
+    private function replacePuck($puck)
+    {
+        $freeCells = $this->getFreeCells();
+        $index = rand(0, count($freeCells) - 1);
```

Comment:
the replacement of the puck is done randomly ?

Reply:
Yes, that the requirement

Reply:
For now. but in the real game it's not random